### ISSUE TITLE
Serentiy Report with protracor and cucumber

### ISSUE THREAD

**Issue by alialrabi** [jan-molak/serenity-js#37]
I have protractor cucumber example works fine, now I'm trying to create report by serenity.js 

protractor.conf.js file

```javascript
exports.config = {
    getPageTimeout: 600000,
    allScriptsTimeout: 700000,
    framework: 'custom',
    frameworkPath: require.resolve('serenity-js'),
    capabilities: {
        'browserName': 'chrome'
    },
    specs: [
        '/home/git/adap_gateway/src/test/features/*.feature'
    ],
    baseURL: 'http://localhost:8099/',
    cucumberOpts: {
        require: '/home/git/adap_gateway/src/test/javascript/stepDefinitions.js',
        format: ['pretty'],
    }
};
```

The feature file 

```gherkin
Feature: Running Cucumber with Protractor

  Scenario: Protractor and Cucumber Test
    Given I go to "http://localhost:8099/#/"
    When I add login credential
    Then I go to scenario home page
    Then I go to scenario details page
    Then I go to edit attack tree page
    Then build attack tree
```
        
ANd package.json file is 

```json
{
    "name": "c",
    "version": "1.0.0",
    "description": "",
    "main": "index.js",
     "dependencies": {
      "gulp": "^3.9.1",
      "protractor-cucumber": "^0.1.8",
     "protractor-cucumber-framework": "^3.1.0"
     },
     "devDependencies": {
     "protractor-cucumber": "^0.1.8",
     "serenity-cli": "^0.2.4",
     "serenity-js": "^1.4.1",
     },
    "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "prereport": "serenity update",
     "report":  "serenity run"
    },
    "author": "",
    "license": "ISC"
}
```

I get 

```
E/launcher - Error: Error: EACCES: permission denied, scandir '/etc/cups/ssl' at Error (native) at Object.fs.readdirSync (fs.js:952:18) at GlobSync._readdir (/home/ali/node_modules/cucumber/node_modules/glob/sync.js:2‌​88:41) [18:15:41] E/launcher - 
Process exited with error code 100,
```

Do you have any instructions?

**Comment by jan-molak** [comment]
Hey @alialrabi,

At the first glance, this looks like a problem caused by [glob](https://www.npmjs.com/package/glob) trying to scan the root directory of your computer.

Is there any reason why the configuration above uses an absolute path to feature files:
```
specs: [
    '/home/git/adap_gateway/src/test/features/*.feature'
],
```
and step definitions:
```
cucumberOpts: {
    require: '/home/git/adap_gateway/src/test/javascript/stepDefinitions.js',
    format: ['pretty'],
}
```
rather than a relative one:
```
specs: [
    'src/test/features/*.feature'
],
```
and
```
cucumberOpts: {
    require: 'src/test/javascript/stepDefinitions.js',
    format: ['pretty'],
}
```
I'm assuming here that the `protractor.conf.js` file lives under `/home/git/adap_gateway/`.

**Comment by alialrabi** [comment]
I have changed to 

       exports.config = {
		  getPageTimeout: 600000,
		  allScriptsTimeout: 700000,
		  framework: 'custom',
		  frameworkPath: require.resolve('serenity-js'), 
		  serenity: {
		        dialect: 'cucumber'     // or 'mocha'
		    },		  
		  capabilities: {
		    'browserName': 'chrome'
		  },
		  specs: [
		    './features/*.feature'
		  ],
		  baseURL: 'http://localhost:8099/',
		  cucumberOpts: {
		    require: './stepDef/stepDefinitions.js'
		  },		  
		  resultJsonOutputFile: 'report.json',
		  ignoreUncaughtExceptions: true 			  
		};

**Comment by jan-molak** [comment]
Sorry to hear that. Would you mind creating a sample project on github demonstrating the issue so that I can try to reproduce the problem?

**Comment by alialrabi** [comment]
Thanks for your help I have created simple jhipster project with protractor cucumber and serenity project https://github.com/alialrabi/serenity.git

**Comment by jan-molak** [comment]
It seems like the [package.json](https://github.com/alialrabi/serenity/blob/master/package.json) has loads of dependencies but is missing some rather important ones, like `serenity-js`, `cucumber` and `protractor`.

Please make sure you follow the installation steps as described in the [Serenity/JS Handbook](http://serenity-js.org/overview/installation.html), and give the [tutorial](http://serenity-js.org/from-scripts-to-serenity/readme.html) a try to see how the different pieces of the puzzle fit together.

**Comment by alialrabi** [comment]
I appreciate your concern, package.json updated  after new installation but the problem still there

       {
  "name": "c",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "dependencies": {
    "gulp": "^3.9.1",
    "protractor-cucumber": "^0.1.8",
    "protractor-cucumber-framework": "^3.1.0"
  },
  "devDependencies": {
    "@types/chai": "^3.5.2",
    "@types/chai-as-promised": "0.0.30",
    "@types/cucumber": "^2.0.1",
    "@types/mocha": "^2.2.41",
    "@types/node": "^7.0.14",
    "chai": "^3.5.0",
    "chai-as-promised": "^6.0.0",
    "chai-smoothie": "^0.2.1",
    "cucumber": "^1.3.3",
    "mocha": "^3.3.0",
    "protractor": "^5.1.1",
    "serenity-cli": "^0.2.4",
    "serenity-js": "^1.4.1",
    "ts-node": "^3.0.2",
    "typescript": "^2.3.2"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}

**Comment by alialrabi** [comment]
When I tried run by super user 

          $  sudo protractor protractor.conf.js 

get this error
 
         [19:04:09] E/launcher - Error: Error: EINVAL: invalid argument, scandir '/proc/11432/net'

**Comment by jan-molak** [comment]
Hmm, what's the reason behind using `sudo` to run protractor? This runs your tests with elevated permissions, I'm not sure if that's what you want?

I guess it might all come down to a problem with your node/npm installation and perhaps using global modules installed as root instead of local ones installed with the correct permissions?

What I'd suggest is:
- see what global modules you have installed, you shouldn't have protractor or cucumber listed there: `npm list -g --depth=0`
- add an [npm script](http://serenity-js.org/from-scripts-to-serenity/making-the-tests-speak-for-themselves.html) to run protractor via npm from the terminal

**Comment by jan-molak** [comment]
@alialrabi did my suggestion help?

**Comment by alialrabi** [comment]
npm list is

             ali@ali:~$ npm list -g --depth=0
             /usr/local/lib
            ├── cucumber@2.0.0-rc.9
            ├── generator-jhipster-elasticsearch-reindexer@0.3.0
            ├── gulp@3.9.1
            ├── gulp-rev@7.1.2
            ├── npm@4.5.0
            ├── pageres@4.4.0
            ├── protractor@5.1.1
            └── version@0.1.2

I have followed https://semaphoreci.com/community/tutorials/getting-started-with-protractor-and-cucumber 

So protractor and cucmber installed globally ,frameworks like protractor-cucumber-framwork and serenity.js installed locally ,Is this Wrong? 

Install npm by npm install -g npm 
Install node by download it from site

**Comment by jan-molak** [comment]
I wouldn't recommend installing dependencies globally (`-g`) for a number of reasons:
- they're not under control of the project that needs them, which often puts teams in a situation where some people have one version of cucumber, karma, protractor or gulp and they're colleagues have a different version, exhibiting different behaviour (and different bugs).
- they can conflict with local dependencies, for example when the global dependency uses local, potentially incompatible plugins.

Installing node using the [regular installer](https://nodejs.org/en/download/) is, of course, fine. Here, I'd only suggest using the [Long-Term Support](https://nodejs.org/en/download/) version instead of the "current" one.

Also, your installation seems to be using a Release Candidate version of Cucumber, which is not yet supported (`2.0.0-rc.9`). Please use the latest stable version instead (`1.x`).
You also won't need the `protractor-cucumber-framework`, as Serenity/JS acts as its replacement.

Installation of Serenity/JS as well as any supporting libraries that you might need is described in the [Serenity/JS tutorial](http://serenity-js.org/from-scripts-to-serenity/readme.html). You can also find the step-by-step instructions in the [documentation](http://serenity-js.org/overview/installation.html).

If the Serenity/JS experiment is the only project you're working on on your computer, I'd suggest uninstalling the global dependencies and following the steps described in the [documentation](http://serenity-js.org/overview/installation.html) instead.

You can uninstall global packages like that:
```
npm uninstall -g protractor cucumber
```

Hope this helps!

**Comment by alialrabi** [comment]
@jan-molak, 
Many Thanks for your clear recommendation ,I will follow it.

**Comment by alialrabi** [comment]
Hi @jan-molak,

How can run protractor locally?

protractor protractor.conf.js  gets    protractor: command not found

gulp protractor get error with webdriver,How to install selenium webdriver locally?

**Comment by jan-molak** [comment]
@alialrabi you can use the [npm scripts](https://docs.npmjs.com/misc/scripts), it's all explained in [the tutorial](http://serenity-js.org/from-scripts-to-serenity/readme.html).

You can also check the Serenity/JS [example projects](https://github.com/jan-molak/serenity-js/blob/master/examples/), you'll notice that I've already set up those npm scripts in the `package.json` file.

For example, [here](https://github.com/jan-molak/serenity-js/blob/8d7edbdefdc35dc24025193c1c047bd46994c496/examples/todomvc-protractor-cucumber/package.json#L11) you'll see that I've created an npm script as follows:

```json
    "e2e": "protractor ./protractor.conf.js",
```

This allows me to run protractor by issuing the following command from the terminal:

```
npm run e2e
```

In short, you can execute any npm script by issuing `npm run <name of the script>`
